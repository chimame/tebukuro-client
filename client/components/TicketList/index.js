import React     from 'react'

const TicketList = ({tickets}) => {
  return (
    <ul>
      {
        tickets.map((ticket) => {
          return (
            <li key={ ticket.id }>
              <p>name: { ticket.name }</p>
              <p>cost: { ticket.cost }</p>
              <p>quantity: { ticket.quantity }</p>
              // TODO: These Date object strings should be formatted.
              <p>saleStartsAt: { ticket.makeDateWith('saleStartsAt').toString() }</p>
              <p>saleEndsAt: { ticket.makeDateWith('saleEndsAt').toString() }</p>
            </li>
          )
        })
      }
    </ul>
  )
}

TicketList.propTypes = {
  tickets: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
}

export default TicketList
