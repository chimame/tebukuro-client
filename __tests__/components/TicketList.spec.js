import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import TicketModel from '../../client/models/Ticket'
import TicketList from '../../client/components/TicketList'

const model = (params) => {
  return new TicketModel(params)
}

describe('TicketList', () => {
  it('render list of tickets', () => {
    const ticketsParams = [
      {
        id:            3,
        name:          'ticketName1',
        cost:          4,
        quantity:      5,
        eventId:       1,
        saleStartsAt: '2017-03-01T09:00:00.000Z',
        saleEndsAt:   '2017-03-01T10:30:00.000Z'
      },
      {
        id:            7,
        name:          'ticketName2',
        cost:          8,
        quantity:      9,
        eventId:       1,
        saleStartsAt: '2017-12-11T13:30:00.000Z',
        saleEndsAt:   '2017-12-11T15:00:00.000Z'
      }
    ]

    const tickets = ticketsParams.map((params) => model(params))

    const component = shallow(<TicketList tickets={ tickets } />)

    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
