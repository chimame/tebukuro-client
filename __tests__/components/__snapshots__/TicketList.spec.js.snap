// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketList render list of tickets 1`] = `
<ul>
  <li>
    <p>
      name: 
      ticketName1
    </p>
    <p>
      cost: 
      4
    </p>
    <p>
      quantity: 
      5
    </p>
    // TODO: These Date object strings should be formatted.
    <p>
      saleStartsAt: 
      Wed Mar 01 2017 18:00:00 GMT+0900 (JST)
    </p>
    <p>
      saleEndsAt: 
      Wed Mar 01 2017 19:30:00 GMT+0900 (JST)
    </p>
  </li>
  <li>
    <p>
      name: 
      ticketName2
    </p>
    <p>
      cost: 
      8
    </p>
    <p>
      quantity: 
      9
    </p>
    // TODO: These Date object strings should be formatted.
    <p>
      saleStartsAt: 
      Mon Dec 11 2017 22:30:00 GMT+0900 (JST)
    </p>
    <p>
      saleEndsAt: 
      Tue Dec 12 2017 00:00:00 GMT+0900 (JST)
    </p>
  </li>
</ul>
`;
